common/workunit/workunit.cpp

```diff
@@ -2657,85 +2660,187 @@ cost_type aggregateCost(const IConstWorkUnit * wu, const char *scope, bool exclu
         }
         return totalCost;
     }
+};
+
+GlobalStatisticCollection::GlobalStatisticCollection() : aggregateKindsMapping(aggregateKindStatistics)
+{
+    // Construct statsCollection here as GlobalStatisticCollection::load() is optional
+    StatsScopeId globalScopeId(SSTglobal, (unsigned)0);
+    statsCollection.setown(createStatisticCollection(nullptr, globalScopeId));
 }
 
-//aggregate disk costs from top-level subgraphs (when scope specified) or workflows (scope not specified)
-cost_type aggregateDiskAccessCost(const IConstWorkUnit * wu, const char *scope)
+void GlobalStatisticCollection::load(IConstWorkUnit &workunit, const char * graphName, bool aggregatesOnly)
 {
-    WuScopeFilter filter;
-    if (!isEmptyString(scope))
-        filter.addScope(scope);
-    else
-        filter.addScope("");      // Needed to match scope
-    // when scope is a workflow, sum graph costs (or subgraph cost when no graph cost) to get workflow cost
-    // (Costs from child graphs and activities should have been summed up to graph/subgraph level already)
-    // when isEmptyString(scope), sum workflow costs (or graph cost when no workflow cost) to get global cost
-    // (Costs from all levels below graph should be summed upto at least graph level already)
-    // i.e. need 2 levels of nesting
-    filter.setIncludeNesting(2);
-    // includeNesting(2) needs just source "global". However, WuScopeFilter is incorrectly inferring the source as "global,stats",
-    // causing too many of the stats to be pulled in and inefficiency.  Here, explicitly set source to "global"
-    filter.addSource("global");
-    filter.addOutputStatistic(StCostFileAccess);
-    filter.addRequiredStat(StCostFileAccess);
-    filter.finishedFilter();
-    Owned<IConstWUScopeIterator> it = &wu->getScopeIterator(filter);
-    cost_type totalCost = 0;
-    for (it->first(); it->isValid(); )
+    const char * _wuid = workunit.queryWuid();
+    if (!streq(_wuid, wuid.str())) // New statsCollection if collection for different workunit
     {
-        cost_type value = 0;
-        if (it->getStat(StCostFileAccess, value))
-        {
-            totalCost += value;
-            it->nextSibling();
-        }
-        else
+        StatsScopeId globalScopeId(SSTglobal, (unsigned)0);
+        statsCollection.setown(createStatisticCollection(nullptr, globalScopeId));
+        wuid.set(_wuid);
+    }
+
+    loadGlobalAggregates(workunit);
+    Owned<IPropertyTree> root = getWUGraphProgress(wuid, true);
```

Comment:
The normal expected usage of GlobalStatisticCollection::load, is that the graphName provided is graph that either hasn't been run, or is being rerun and it's stats. previously clearer (if not cleared, the aggregates would be confusing..).

I think should elude to that here, to say that there are generally not expected to be any subgraph stats from graphName here.

I wonder whether the engines should pass in graphName=nullptr , to explicitly tell it not to even bother to try to load. Which would save a call + lock to dali on /GraphProgress which isn't going to be used.